base/applications/osk/main.c

```diff
@@ -58,6 +59,32 @@ int OSK_SetImage(int IdDlgItem, int IdResource)
     return TRUE;
 }
 
+/***********************************************************************
+ *
+ *           OSK_SetImageApp
+ *
+ *  Set the application's icon
+ */
+BOOL OSK_SetImageApp()
+{
+    HICON hIconSmall;
+
+    /* Load the icon */
+    hIconSmall = LoadIconW(Globals.hInstance, MAKEINTRESOURCE(IDI_OSK));
```

Comment:
LoadIcon is deprecated. See MSDN.

Reply:
Do you have any ideas why `LoadImageW()` fails to take the relative path of an icon and succeeds if you specify the absolute path? Like doing so for example:
```C
hIconSmall = LoadImageW(NULL, L"E:\\ReactOS\\Fork\\reactos\\base\\applications\\osk\\res\\osk.ico", IMAGE_ICON, 0, 0, LR_DEFAULTSIZE | LR_LOADFROMFILE);
```
And that loads the icon without problems. But if I set `res\\osk.ico` it fails and sets the default icon.

Reply:
Because `res\osk.ico` is just a path in the source tree, which is not valid inside built ReactOS.

Use `LoadImageW` the same way as you've done with `LoadIconW`.

See https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=LoadImageW

Reply:
https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/user/user32/windows/cursoricon.c;hb=de7c959c4e5f287c81e61c7e369a5f0ae2c33398#l2044 shows you how LoadIconW is implemented, so you can deduce the parameters that will result in the same behavior as your LoadIconW call.